src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -277,12 +329,83 @@ private AzureFileSystem GetAzureFileSystem()
             return fileSystem;
         }
 
-        private async Task<bool> PushAsync(IEnumerable<string> items, bool allowOverwrite)
+        private async Task<bool> PushAsync(
+            IEnumerable<string> items,
+            bool allowOverwrite,
+            bool passIfExistingItemIdentical)
         {
             LocalSettings settings = GetSettings();
             AzureFileSystem fileSystem = GetAzureFileSystem();
-            bool result = await PushCommand.RunAsync(settings, fileSystem, items.ToList(), allowOverwrite, skipExisting: false, log: new SleetLogger(Log));
-            return result;
+            SleetLogger log = new SleetLogger(Log);
+
+            var packagesToPush = items.ToList();
+
+            if (!allowOverwrite && passIfExistingItemIdentical)
+            {
+                var context = new SleetContext
+                {
+                    LocalSettings = settings,
+                    Log = log,
+                    Source = fileSystem,
+                    Token = CancellationToken
+                };
+                context.SourceSettings = await FeedSettingsUtility.GetSettingsOrDefault(
+                    context.Source,
+                    context.Log,
+                    context.Token);
+
+                var flatContainer = new FlatContainer(context);
+
+                var packageIndex = new PackageIndex(context);
+
+                // Check packages sequentially: Task.WhenAll caused IO exceptions in Sleet.
+                for (int i = packagesToPush.Count - 1; i >= 0; i--)
+                {
+                    string item = packagesToPush[i];
+
+                    bool? identical = await IsPackageIdenticalOnFeedAsync(
+                        item,
+                        packageIndex,
+                        context.Source,
+                        flatContainer,
+                        log);
+
+                    if (identical == null)
+                    {
+                        continue;
+                    }
+
+                    packagesToPush.RemoveAt(i);
+
+                    if (identical == true)
+                    {
+                        Log.LogMessage(
+                            MessageImportance.Normal,
+                            "Package exists on the feed, and is verified to be identical. " +
+                            $"Skipping upload: '{item}'");
+                    }
+                    else if (identical == false)
```

Comment:
nit: just `else`

Reply:
This is a `bool?`, so that would change things. Kind of the classic `True` `False` `FileNotFound` scenario. I'll change this to an enum to clear it up--`bool?` doesn't have enough meaning to make this readable.

Reply:
`identical` can't be null at this point because of the `if (identical == null) { continue; }` line above.

Reply:
Oh, right. :smile: Thanks.